packages/core/src/store/anchor-request-store.ts

```diff
@@ -98,12 +106,22 @@ export class AnchorRequestStore extends ObjectStore<StreamID, AnchorRequestData>
     let numEntries = 0
     do {
       try {
-        const batch = await this.store.find({
+        let timeout
+        const timeoutPromise = new Promise<null>((resolve) => {
+          timeout = setTimeout(() => {
+            this.#logger.warn(`Timed out while waiting for AnchorRequestStore to fetch a batch`)
+            resolve(null)
+          }, this.#infiniteListBatchTimeoutMs)
+        })
+
```

Comment:
Can we add a debug logs here? before initiating a find operation, this log might be beneficial in cases where we think the find operation hangs, we should add a subsequent completion log as well when the operations completes. Mostly when we check batch is non null and length is greater than 0
 
this.#logger.debug(`Initiating 'find' operation with batchSize: ${batchSize} and gt: ${gt}`);
```suggestion
```suggestion
this.#logger.debug(`Initiating 'find' operation in with batchSize: ${batchSize} and gt: ${gt}`);
```
```

Reply:
I'm worried that will be very verbose and will fill up our logs.  I'd like to try this one as-is first, and if we still can't figure it out after this PR goes in, then we can try a build with super verbose logging tracking every single part of the polling process, and revert it after we figure out what's wrong.